app/src/main/kotlin/io/intrepid/skotlinton/utils/LiveDataObserver.kt

```diff
@@ -0,0 +1,27 @@
+package io.intrepid.skotlinton.utils
+
+import android.view.View
+import android.widget.TextView
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.Observer
+
+/**
+ * Interface for common utility methods used by LifecyleOwners (Activities and Fragments) relating to LiveData
+ */
+interface LiveDataObserver {
+
+    val liveDataLifecycleOwner: LifecycleOwner
+
+    fun <T> LiveData<T>.observe(observer: (T) -> Unit) {
+        observe(liveDataLifecycleOwner, Observer { observer(it) })
+    }
+
+    fun LiveData<Boolean>.bindToVisibility(view: View) {
+        observe { visible -> visible.let { view.visibility = if (it) View.VISIBLE else View.GONE } }
+    }
+
+    fun LiveData<String>.bindToText(view: TextView) {
```

Comment:
The main intent here is to reduce boilerplate. There would just be bindTo methods for the most common use cases of liveData. I don't think the list would be large. Off the top of my head, it would just be checked state, enabled state, and image/background. It's up to the project on what they want to include here.

Reply:
Ok I can get behind that in general. On second pass here, though, I think for most projects having this defined on `String` may be an antipattern since text should be derived from resources. But that's probably fine here for demonstration purposes.